Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -208,6 +264,74 @@ void DiscusMultipleScatteringCorrection::getXMinMax(const Mantid::API::MatrixWor
   }
 }
 
+void DiscusMultipleScatteringCorrection::prepareStructureFactor() {
+  // avoid repeated conversion of bin edges to points inside loop by converting to point data
+  convertWsToPoints(m_SQWS);
+  // if S(Q,w) has been supplied ensure Q is along the x axis of each spectrum (so same as S(Q))
+  if (m_SQWS->getAxis(1)->unit()->unitID() == "MomentumTransfer") {
+    auto transposeAlgorithm = this->createChildAlgorithm("Transpose");
+    transposeAlgorithm->initialize();
+    transposeAlgorithm->setProperty("InputWorkspace", m_SQWS);
+    transposeAlgorithm->setProperty("OutputWorkspace", "_");
+    transposeAlgorithm->execute();
+    m_SQWS = transposeAlgorithm->getProperty("OutputWorkspace");
+  } else if (m_SQWS->getAxis(1)->isSpectra()) {
+    // for elastic set w=0 on the spectrum axis to align code with inelastic
+    auto newAxis = std::make_unique<NumericAxis>(std::vector<double>{0.});
+    newAxis->setUnit("DeltaE");
+    m_SQWS->replaceAxis(1, std::move(newAxis));
+  }
+  // avoid repeated conversion of bin edges to points inside loop by converting to point data
+  convertWsToPoints(m_SQWS);
```

Comment:
This this is repeated a few lines before (L269)

Reply:
It's supposed to be called again if the axes on the workspace have been transposed so that both axes (Q,w) are converted to points. But it's not in the right place. Should be at the end of the first bit of the if...else statement so I'll move it there.

Reply:
Just noticed that convertWSToPoints actually converts both axis (x axis and spectrum axis) so no need to call it twice at all. I'll remove the second call